teamcity: failed tests on master: test (proposer evaluated kv)/TestPushTxnHeartbeatTimeout

The following tests appear to have failed:

[#193266](https://teamcity.cockroachdb.com/viewLog.html?buildId=193266):

```
--- FAIL: test (proposer evaluated kv)/TestPushTxnHeartbeatTimeout (0.010s)
replica_test.go:4387: 3: expSuccess=false; got pErr <nil>, reply header:<num_keys:0 > pushee_txn:<meta:<id:<f6bd9663-cd89-4cb8-8cf7-e87834b6195f> isolation:SERIALIZABLE key:"key-3" epoch:0 timestamp:<wall_time:5000000123 logical:26 > priority:0 sequence:1 batch_index:0 > name:"test-3" status:PENDING orig_timestamp:<wall_time:123 logical:53 > max_timestamp:<wall_time:124 logical:53 > writing:true write_too_old:false retry_on_push:false > 
------- Stdout: -------
I170324 15:14:03.833121 5121 storage/store.go:1318  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170324 15:14:03.836126 5121 util/stop/stopper.go:493  quiescing; tasks left:
2      gossip/infostore.go:301
I170324 15:14:03.836160 5121 util/stop/stopper.go:493  quiescing; tasks left:
1      gossip/infostore.go:301

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/roachpb/data.go
+++ b/pkg/roachpb/data.go
@@ -818,8 +818,10 @@ func MakePriority(userPriority UserPriority) int32 {
 	// For userPriority=MaxUserPriority, the probability of overflow is 0.7%.
 	// For userPriority=(MaxUserPriority/2), the probability of overflow is 0.005%.
 	val = (val / (5 * MaxUserPriority)) * math.MaxInt32
-	if val >= MaxTxnPriority {
-		return MaxTxnPriority
+	if val <= MinTxnPriority {
+		return MinTxnPriority + 1
+	} else if val >= MaxTxnPriority {
+		return MaxTxnPriority - 1
 	}
 	return int32(val)
 }

--- a/pkg/storage/replica_test.go
+++ b/pkg/storage/replica_test.go
@@ -4524,12 +4524,9 @@ func TestPushTxnHeartbeatTimeout(t *testing.T) {
 		{hlc.Timestamp{}, ns * 2, roachpb.PUSH_TIMESTAMP, false},
 		{hlc.Timestamp{}, ns * 2, roachpb.PUSH_ABORT, false},
 		{hlc.Timestamp{}, ns * 2, roachpb.PUSH_TOUCH, false},
-		{ts, ns*2 + 1, roachpb.PUSH_TIMESTAMP, false},
-		{ts, ns*2 + 1, roachpb.PUSH_ABORT, false},
-		{ts, ns*2 + 1, roachpb.PUSH_TOUCH, false},
-		{ts, ns*2 + 2, roachpb.PUSH_TIMESTAMP, true},
-		{ts, ns*2 + 2, roachpb.PUSH_ABORT, true},
-		{ts, ns*2 + 2, roachpb.PUSH_TOUCH, true},
+		{ts, ns*2 + 1, roachpb.PUSH_TIMESTAMP, true},
+		{ts, ns*2 + 1, roachpb.PUSH_ABORT, true},
+		{ts, ns*2 + 1, roachpb.PUSH_TOUCH, true},
 	}
 
 	for i, test := range testCases {
@@ -4542,22 +4539,22 @@ func TestPushTxnHeartbeatTimeout(t *testing.T) {
 			pushee.LastHeartbeat = test.heartbeat
 		}
 		_, btH := beginTxnArgs(key, pushee)
-		btH.Timestamp = tc.repl.store.Clock().Now()
+		btH.Timestamp = pushee.Timestamp
 		put := putArgs(key, key)
 		if _, pErr := maybeWrapWithBeginTransaction(context.Background(), tc.Sender(), btH, &put); pErr != nil {
 			t.Fatalf("%d: %s", i, pErr)
 		}
 
-		// Now, attempt to push the transaction with Now set to our current time.
+		// Now, attempt to push the transaction with Now set to the txn start time + offset.
 		args := pushTxnArgs(pusher, pushee, test.pushType)
-		args.Now = now.Add(test.timeOffset, 0)
+		args.Now = pushee.Timestamp.Add(test.timeOffset, 0)
 		args.PushTo = args.Now
 
 		reply, pErr := tc.SendWrapped(&args)
 
 		if test.expSuccess != (pErr == nil) {
-			t.Fatalf("%d: expSuccess=%t; got pErr %s, reply %+v", i,
-				test.expSuccess, pErr, reply)
+			t.Fatalf("%d: expSuccess=%t; got pErr %s, args=%+v, reply=%+v", i,
+				test.expSuccess, pErr, args, reply)
 		}
 		if pErr != nil {
 			if _, ok := pErr.GetDetail().(*roachpb.TransactionPushError); !ok {

